src/Processors/Formats/IRowOutputFormat.cpp

```diff
@@ -21,12 +21,12 @@ void IRowOutputFormat::consume(DB::Chunk chunk)
     {
         if (!first_row)
             writeRowBetweenDelimiter();
-        first_row = false;
 
         write(columns, row);
+        first_row = false;
```

Comment:
The place where `first_row = false` is written doesn't look logical.
It seems it would be better to move `first_row = false` to the end, after calling `params.callback`.